cmd/kube-proxy/app/server.go

```diff
@@ -623,6 +623,10 @@ func (s *ProxyServer) Run() error {
 	endpointsConfig.RegisterEventHandler(s.Proxier)
 	go endpointsConfig.Run(wait.NeverStop)
 
+	nodeConfig := config.NewNodeConfig(informerFactory.Core().V1().Nodes(), s.ConfigSyncPeriod)
+	nodeConfig.RegisterEventHandler(s.Proxier)
+	go nodeConfig.Run(wait.NeverStop)
```

Comment:
Should the node watch also be feature gated?

Reply:
yes, I have found it, planning add a commit for this tomorrow morning :)

Reply:
added a commit, watch node only if `TopologyAwareServiceRouting` is enabled